### PR TITLE
[#40] [Backend] As a user, I can submit a survey

### LINKED ISSUE #40: [Backend] As a user, I can submit a survey
## Why

User must be able to submit a survey after answering all the questions

## Acceptance Criteria

- Make post request to `https://survey-api.nimblehq.co/api/v1/responses` and handle the result
- If successful, return success
- If unsuccessful, return the error message from API

## Resources

- [API](https://documenter.getpostman.com/view/11835486/UVsPNPfU#8e233fe5-b91e-4b98-83fc-ad3c98775180)

### PR DESCRIPTION
**Pull request by ryan-conway**
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/40

## What happened 👀

- Created `SubmitSurveyUseCase` and relevant api classe

## Insight 📝

N/A

## Proof Of Work 📹

```
I/flutter (19984): *** Request ***
I/flutter (19984): uri: https://survey-api.nimblehq.co/api/v1/responses
I/flutter (19984): method: POST
I/flutter (19984): responseType: ResponseType.json
I/flutter (19984): followRedirects: true
I/flutter (19984): connectTimeout: 3000
I/flutter (19984): sendTimeout: 0
I/flutter (19984): receiveTimeout: 5000
I/flutter (19984): receiveDataWhenStatusError: true
I/flutter (19984): extra: {}
I/flutter (19984): headers:
I/flutter (19984):  content-type: application/json; charset=utf-8
I/flutter (19984):  Authorization: Bearer Jo32kFbWS1uwFsDMJbUuDHGNv5gqXdpIBo_Ryu9-mpY
I/flutter (19984): data:
I/flutter (19984): Instance of 'SubmitSurveyRequest'
I/flutter (19984): 
I/flutter (19984): *** Response ***
I/flutter (19984): uri: https://survey-api.nimblehq.co/api/v1/responses
I/flutter (19984): statusCode: 201
I/flutter (19984): headers:
I/flutter (19984):  connection: keep-alive
I/flutter (19984):  cache-control: max-age=0, private, must-revalidate
I/flutter (19984):  date: Wed, 19 Jul 2023 07:57:50 GMT
I/flutter (19984):  vary: Accept-Encoding, Origin
I/flutter (19984):  content-encoding: gzip
I/flutter (19984):  referrer-policy: strict-origin-when-cross-origin
I/flutter (19984):  report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=oNc1xvT3BGgNOnRaRJ3axj2P8vxPPXAqD1sCh3Vz7jZvp7hwftiX9oUYYMiZ3icb1x8RnRfPtHLF5sE9QtIcF1I18%2BmJRfxeuZfcDPZ33BNCIYrUjBErqa3aKRPc3KGOAVJf6HeA2bDJu1dRj5Jmp3pmojZj"}],"group":"cf-nel","max_age":604800}
I/flutter (19984):  cf-cache-status: DYNAMIC
I/flutter (19984):  x-permitted-cross-domain-policies: none
I/flutter (19984):  content-type: application/json; charset=utf-8
I/flutter (19984):  x-xss-protection: 0
I/flutter (19984):  server: cloudflare
I/flutter (19984):  x-request-id: e29311d3-5125-47ec-b8c6-e6560c0b3b15
I/flutter (19984):  alt-svc: h3=":443"; ma=86400
I/flutter (19984):  content-length: 28
I/flutter (19984):  nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
I/flutter (19984):  x-download-options: noopen
I/flutter (19984):  cf-ray: 7e916a7289be4b6a-BKK
I/flutter (19984):  x-runtime: 0.006749
I/flutter (19984):  etag: W/"bfc56c7573d885cbc6723c948334098c"
I/flutter (19984):  via: 1.1 vegur
I/flutter (19984):  x-frame-options: SAMEORIGIN
I/flutter (19984):  x-content-type-options: nosniff
I/flutter (19984): Response Text:
I/flutter (19984): {}
I/flutter (19984): 
```

### CODE REVIEW

**Review by markgravity — APPROVED** [review]

File: lib/model/request/submit_survey_answer_request.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'submit_survey_answer_request.g.dart';
+
+@JsonSerializable()
+class SubmitSurveyAnswerRequest {
```
Comment (lydiasama) [inline lib/model/request/submit_survey_answer_request.dart]: Can we name it `AnswerRequest` like we have `AnswerResponse` that is quite similar?

**Review by lydiasama — COMMENTED** [review]
Just one small thing. The rest LGTM. 👍🏻

File: lib/model/request/submit_survey_answer_request.dart (hunk 1 above)
Comment (ryan-conway) [reply lib/model/request/submit_survey_answer_request.dart]: Updated in [55e6acc](https://github.com/nimblehq/flutter-ic-lydia-ryan/pull/109/commits/55e6acc61c4bfdf9b96b54abeaaf3431243af4fa)

**Review by manh-t — APPROVED** [review]
LGTM 👍

**Pull request merged**